content/blog/blog-2020-07-27-building-your-first-operator-3.md

```diff
@@ -0,0 +1,437 @@
+
+```yaml
+       - name: cleanup
+            tasks:
+              - bootstrap_cleanup
+              - config
+
+    - name: config
+      kind: Apply
+      spec:
+        resources:
+          - galera_config.yaml
+```
+
+We’ve added two tasks to a single step here. These will be executed serially, since in our deploy phase we have configured the execution strategy as serial.
+
+```yaml
+    phases:
+      - name: deploy
+        strategy: serial
+```
+
+Notice that also we’re referring to the pre-existing `galera_config.yaml` that we used in Part 2 to configure our nodes. In our templates, we also have access to a few other system [variables](../docs/developing-operators/templates.md#variables), one of which is `.StepName`
+
+We can use that variable to insert conditionals into our templates depending on which step they were called in, so let’s modify that file :
+
+```yaml
+# galera_config.yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ .Name }}-nodeconfig
+  namespace: {{ .Namespace }}
+data:
+  galera.cnf: |
+    [galera]
+    wsrep_on = ON
+    wsrep_provider = /usr/lib/galera/libgalera_smm.so
+    wsrep_sst_method = mariabackup
+    {{ if eq .StepName "firstboot_config" -}}
+    wsrep_cluster_address = gcomm://{{ .Name }}-bootstrap-svc,{{ $.Name }}-{{ $.OperatorName }}-0.{{ $.Name }}-hs{{ range $i, $v := untilStep 1 (int .Params.NODE_COUNT) 1 }},{{ $.Name }}-{{ $.OperatorName }}-{{ $v }}.{{ $.Name }}-hs{{end}}
+    {{ else -}}
+    wsrep_cluster_address = gcomm://{{ $.Name }}-galera-0.{{ $.Name }}-hs{{ range $i, $v := untilStep 1 (int .Params.NODE_COUNT) 1 }},{{ $.Name }}-galera-{{ $v }}.{{ $.Name }}-hs{{end}}
+    {{ end -}}
+    wsrep_sst_auth = "root:{{ .Params.MYSQL_ROOT_PASSWORD }}"
+    binlog_format = ROW
+  innodb.cnf: |
+    [innodb]
+    innodb_autoinc_lock_mode = 2
+    innodb_flush_log_at_trx_commit = 0
+    innodb_buffer_pool_size = 122M
+```
+
+You can see we’ve added the conditional step around the `wsrep_cluster_address`, which means it gets the bootstrap configuration if called in the `firstboot_config` step, otherwise it just has the nodes in the cluster. 
+
+```yaml
+{{ if eq .StepName "firstboot_config" -}}
+    wsrep_cluster_address = gcomm://{{ .Name }}-bootstrap-svc,{{ $.Name }}-{{ $.OperatorName }}-0.{{ $.Name }}-hs{{ range $i, $v := untilStep 1 (int .Params.NODE_COUNT) 1 }},{{ $.Name }}-{{ $.OperatorName }}-{{ $v }}.{{ $.Name }}-hs{{end}}
+    {{ else -}}
+    wsrep_cluster_address = gcomm://{{ $.Name }}-galera-0.{{ $.Name }}-hs{{ range $i, $v := untilStep 1 (int .Params.NODE_COUNT) 1 }},{{ $.Name }}-galera-{{ $v }}.{{ $.Name }}-hs{{end}}
+    {{ end -}}
+```
+
+Note that we terminate both the `if` and `end` statements with a period. This tells the templating engine not to leave an empty line where those statements were in the template. For a full explanation of the templating code, have a look at [Part 2](blog-2020-07-building-your-first-operator-2.md) of this blog.
```

Comment:
```suggestion
Note that we terminate both the `if` and `end` statements with a dash. This tells the templating engine not to leave an empty line where those statements were in the template. For a full explanation of the templating code, have a look at [Part 2](blog-2020-07-building-your-first-operator-2.md) of this blog.
```

Or hyphen? Not sure...

Reply:
Good catch, it should actually be if else and end, and hyphen is correct. Amended.